src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -486,6 +490,16 @@ private boolean preBuild(BuildListener listener, List<BuildStep> steps) {
     public static final PermissionGroup PERMISSIONS = new PermissionGroup(Promotion.class, Messages._Promotion_Permissions_Title());
     public static final Permission PROMOTE = new Permission(PERMISSIONS, "Promote", Messages._Promotion_PromotePermission_Description(), Jenkins.ADMINISTER, PermissionScope.RUN);
 
+    @Initializer(before = InitMilestone.SYSTEM_CONFIG_LOADED)
+    @Restricted(DoNotUse.class)
+    public static void registerPermissions() {
+    // Pending JENKINS-17200, ensure that the above permissions have been registered prior to
+    // allowing plugins to adapt the system configuration, which may depend on these permissions
+    // having been registered. Since this method is static and since it follows the above
+    // construction of static permission objects (and therefore their calls to
+    // PermissionGroup#register), there is nothing further to do in this method.
```

Comment:
As a matter of style, I like to do something like

```java
PROMOTE.hashCode();
```

though SpotBugs is likely to complain either way.